src/Sidebar/Sidebar.tsx

```diff
@@ -8,12 +7,43 @@ import {
   SidebarHeaderStyles,
 } from "./sidebar.style";
 
-const renderableChildrenPropType = PropTypes.oneOfType([
-  PropTypes.arrayOf(PropTypes.node),
-  PropTypes.node,
-]);
+type LinkProps = {
+  iconName: string;
+  active?: boolean;
+  href: string;
+  children?: ReactNode;
+};
+
+type HeaderProps = {
+  logoUrl?: string;
+  logoAltText?: string;
+  href?: string;
+  children?: ReactNode;
+};
+
+type SideBarProps = {
+  children: ReactNode;
+};
+
+interface LinkFC<T> extends React.FC<T> {
+  displayName: string;
+}
 
-const Link = ({ iconName, active, href, children }) => {
+interface HeaderFC<T> extends React.FC<T> {
+  displayName: string;
+}
+
+interface SideBarFC<Link, Header, Sidebar> extends React.FC<Sidebar> {
+  Link: LinkFC<Link>;
+  Header: HeaderFC<Header>;
+}
+
+const Link: LinkFC<LinkProps> = ({
```

Comment:
I think it's better to separate those into different files
@Weetbix wdyt?

Reply:
I think it's alright here, better for now to avoid any unnecessary refactoring to keep PRs smaller and changes easier to review etc